### PR TITLE
add android scenarios

### PR DESCRIPTION
**Pull request by billwert**
This adds the first couple android size scenarios. Now that this plumbing is in place I'll work on adding more.

### CODE REVIEW

File: eng/common/performance/performance-setup.ps1 [hunk 1]
```diff
@@ -102,6 +103,15 @@ if ($UseBaselineCoreRun) {
     Move-Item -Path $BaselineCoreRootDirectory -Destination $NewBaselineCoreRoot
 }
 
+if ($AndroidMono) {
+    if(!(Test-Path $WorkItemDirectory))
+    {
+        mkdir $WorkItemDirectory
+    }
+    Copy-Item -path "$SourceDirectory\artifacts\bin\AndroidSampleApp\arm64\Release\android-arm64\publish\apk\bin\HelloAndroid.apk" $PayloadDirectory
```
Comment (marek-safar) [inline eng/common/performance/performance-setup.ps1:111]: We should target the right functional tests subset a sample

/cc @steveisok

Comment (billwert) [reply eng/common/performance/performance-setup.ps1:111]: Happy to fix whatever this means. :)

Of note: I'm hardcoding the architecture here to work around the fact that our Android devices are connected to Windows machines, so it's just easier to run this phase of the pipeline (for building all the rest of our tools) on Windows x64.

This is just the first test - there are more scenarios recently online that @CoffeeFlux has told me about. This PR is really about ensuring we can flow APKs into the perf infrastructure reasonably.

File: eng/pipelines/coreclr/templates/build-perf-sample-apps.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  displayName: ''
+  artifactName: ''
+  archiveExtension: ''
+  archiveType: ''
+  tarCompression: ''
+
+steps:
+# Build Android sample app
+  - ${{ if eq(parameters.osGroup, 'Android') }}:
+    - script: make run MONO_ARCH=arm64 DEPLOY_AND_RUN=false
```
Comment (steveisok) [inline eng/pipelines/coreclr/templates/build-perf-sample-apps.yml:23]: If you switch over to the functional tests, you can simply build them via:

`./dotnet.sh build src/tests/FunctionalTests/Android/Device_Emulator/JIT /p:TargetOS=Android /p:TargetArchitecture=<arch> /p:Configuration=Release`

File: eng/common/performance/performance-setup.ps1 (hunk 1 above)
Comment (billwert) [reply eng/common/performance/performance-setup.ps1:111]: Got it. I am using the AndroidSampleApp as it's what @CoffeeFlux and @SamMonoRT had pointed me to.

These build to very different sizes for me. 

```
PS /runtime> du -h ./emulator/
7.8M    ./emulator/assets
18M     ./emulator/lib/arm64-v8a
18M     ./emulator/lib
16K     ./emulator/META-INF
26M     ./emulator/
PS /runtime> du -h ./sample/
540K    ./sample/assets
4.1M    ./sample/lib/arm64-v8a
4.1M    ./sample/lib
16K     ./sample/META-INF
4.6M    ./sample/
```

It looks like the test sample is maybe picking up a `libmonodroid.so` with symbols or something in it?
```
PS /runtime> ls -lh ./emulator/lib/arm64-v8a/
total 18M
-rw-rw-rw- 1 root root  17M Jan  1  1980 libmonodroid.so
-rw-rw-rw- 1 root root 758K Jan  1  1980 libSystem.IO.Compression.Native.so
-rw-rw-rw- 1 root root  75K Jan  1  1980 libSystem.Native.so
-rw-rw-rw- 1 root root 131K Jan  1  1980 libSystem.Security.Cryptography.Native.OpenSsl.so
PS /runtime> ls -lh ./sample/lib/arm64-v8a/
total 4.1M
-rw-rw-rw- 1 root root 3.1M Jan  1  1980 libmonodroid.so
-rw-rw-rw- 1 root root 758K Jan  1  1980 libSystem.IO.Compression.Native.so
-rw-rw-rw- 1 root root  75K Jan  1  1980 libSystem.Native.so
-rw-rw-rw- 1 root root 131K Jan  1  1980 libSystem.Security.Cryptography.Native.OpenSsl.so
```

These are built off the same sync from last week bc82161670633b6caecdd18a77b01b55dc4bc86a

Comment (billwert) [reply eng/common/performance/performance-setup.ps1:111]: Just to be sure I synced and rebuilt and get a similar result:
```
-rw-r--r-- 1 root root 14M Mar 30 21:47 /runtime/artifacts/bin/Android.Device_Emulator.JIT.Test/net6.0-Release/android-arm64/AppBundle/bin/Android.Device_Emulator.JIT.Test.apk
PS /runtime> ls -lh /runtime/artifacts/bin/AndroidSampleApp/arm64/Release/android-arm64/publish/apk/bin/HelloAndroid.apk
-rw-r--r-- 1 root root 2.3M Mar 30 21:33 /runtime/artifacts/bin/AndroidSampleApp/arm64/Release/android-arm64/publish/apk/bin/HelloAndroid.apk
```

Now I'm at b23205f59ab3853e4b6b43f91a7ca3900c0cc7d0

Comment (marek-safar) [reply eng/common/performance/performance-setup.ps1:111]: @steveisok ^ this looks like we are not running tests with linker

Comment (billwert) [reply eng/common/performance/performance-setup.ps1:111]: @marek-safar @steveisok I took a swing at adding trimming based on the AndroidSampleApp project file and it didn't seem to have an impact, so I'm probably doing something wrong.

If it's okay with you I'd like to just merge this PR so we can finish getting this data flowing. We can add this test when I add other Xamarin apps.

Comment (steveisok) [reply eng/common/performance/performance-setup.ps1:111]: I'm ok with moving forward as long as we can try to follow up.  I think the functional tests are carrying infrastructure related bits that aren't being 100% used.

**Review by DrewScoggins — APPROVED** [review]

File: eng/common/performance/performance-setup.ps1 (hunk 1 above)
Comment (billwert) [reply eng/common/performance/performance-setup.ps1:111]: @steveisok @SamMonoRT It sounds like this is more followup from the functional point of view than the perf point of view. Does one of you want to open an issue to track testing with trimming enabled?

**Pull request merged**